### PR TITLE
Feature/post creation function

### PR DESCRIPTION
**Pull request by moj124**
## Changes

- Rebuild Post and Dog Entitiy relations.
- Re-added the forms required to add a Post object to database correctly.
- Nuclear reset button on the migrations_version database, didn't understand that new mirgrations aren't duplicate tables. Had to wipe this clean with a blank migration state.
- Test command added.

### CODE REVIEW

File: src/Entity/Post.php [hunk 1]
```diff
@@ -49,12 +49,12 @@ public function getId(): ?int
...
     }
 
-    public function setDogId(?dog $dog_id): self
+    public function setDogId(?Dog $dog): self
     {
-        $this->dog_id = $dog_id;
+        $this->dog = $dog;
 
         return $this;
     }
```
Comment (Captainmango) [inline src/Entity/Post.php:60]: Shouldn't these be setting an ID? If they are setting Dog type fields on the Post entity, the name should reflect that. Although, I feel that the id is much more relevant to the Post

File: tests/Controller/FeedControllerTest.php [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+        $post->setDogId($dog->getId());
+        $post->setPostText("Woof Woof");
+        $post->setCreatedAt(new \DateTimeImmutable());
+
+        $postRepo->savePost($post);
+        echo($dog);
+        // $this->assertTrue($newDog->getUserIdentifier() === 'test123');
+        // $client->followRedirects(true);
+    }
```
Comment (Captainmango) [inline tests/Controller/FeedControllerTest.php:36]: Loving the test. is there not a way to do account assumption in Symfony? That would be much more realistic in respect to what you're actually testing.

Comment (Captainmango) [reply tests/Controller/FeedControllerTest.php:36]: I'd also think about the name. Does the name actually give a fair description as to what your test is checking?

**Review by Captainmango — COMMENTED** [review]
Minor things. Otherwise looking good. 👍

File: tests/Controller/FeedControllerTest.php (hunk 2 above)
Comment (Captainmango) [reply tests/Controller/FeedControllerTest.php:36]: have a read of this https://symfony.com/doc/current/testing.html#logging-in-users-authentication

File: src/Entity/Post.php (hunk 1 above)
Comment (moj124) [reply src/Entity/Post.php:60]: Yeah, definitely. I quite tricky to find all the things that need changing, so left as it is and just fed it what waas defined.

Comment (Captainmango) [reply src/Entity/Post.php:60]: We'll refactor this tomorrow and talk more about what is happening here 👍

**Pull request merged**